lib/tbot/bot/bot.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bot
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/client/webclient"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+)
+
+// B is an interface covering various public Bot methods to circumvent
+// import cycle issues.
+type B interface {
```

Comment:
Hmm, I'm always a little uncomfortable with shared interfaces like this. It's a pattern that's currently extensively used in the Teleport codebase, but goes against most of the Go fundementals I'm familiar with. It's preferable imo for each consumer to define its own interface, even if this results in some duplication.